src/client/array/dc_array.c

```diff
@@ -1770,7 +1773,7 @@ dc_array_io(daos_handle_t array_oh, daos_handle_t th,
 	 */
 	if (op_type == DAOS_OPC_ARRAY_READ && array->byte_array) {
 		if (head == NULL) {
-			tse_task_complete(stask, rc);
+			tse_task_complete(stask, rc = 0);
```

Comment:
This line looks a little odd, what exactly is it trying to do?  Mind you rc isn't used later in this function so perhaps just pass 0 to the function?
```suggestion
			rc = 0;
			tse_task_complete(stask, rc);
```